support/globalconfig/infrastructure.go

```diff
@@ -21,18 +21,26 @@ func InfrastructureConfig() *configv1.Infrastructure {
 
 func ReconcileInfrastructure(infra *configv1.Infrastructure, hcp *hyperv1.HostedControlPlane) {
 
+	platformType := hcp.Spec.Platform.Type
+	switch hcp.Spec.Platform.Type {
+	// In case of KubeVirtPlatform, set the Infrastructure to NonePlatform
```

Comment:
can you elaborate here? which mcs error and why?

Reply:
A few years ago there was an effort to make an ipi installer for KubeVirt. This was abandoned, however some logic related to the kubevirt platform remains in the mco/mcs. Unfortunately the left over IPI logic related to kubevirt causes the mcs not to properly serve up ignition when the mcs/mco detects the KubeVirt platform on the `infrastructure` object instead of the previous `none` object.

We will sort through this on the mcs side eventually, for now this issue blocks us from being able to transition to NodePort unless we do an override here for the infrastructure object and keep using `none` when kubevirt is in use.

Reply:
Thanks! make sense, could you please elaborate that a bit in the `// comment`? e.g "the mcs won't serve ignition when configv1.InfrastructureStatus.PlatformStatus is kubevirt so we set it to None" and point to the relevant issue tracking this?